mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -22,66 +22,196 @@
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord record = getStatistics(transaction);
+
+        Integer existingValue = record.getDeaths();
+        Integer newValue = existingValue + amount;
+
+        record.setKills(newValue);
+        record.store(KITPVP_STATISTICS.DEATHS);
+        this.deaths = newValue;
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Adds assists to the user account. Infallible.
+     * @param transaction represents the transaction
+     * @param amount represents the amount of assists to add
+     * @return a result with the new value of assists
+     */
+    public StatisticResult addAssists(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord record = getStatistics(transaction);
+
+        Integer existingValue = record.getAssists();
+        Integer newValue = existingValue + amount;
+
+        record.setKills(newValue);
+        record.store(KITPVP_STATISTICS.ASSISTS);
+
+        this.assists = newValue;
+
+        return new StatisticResult(newValue);
+    }
+
+    //TODO verify if the record-counting part of this (res != 0) is accurate or needs to be flipped
+
+    /**
+     * Adds an existing kit to the player's kit list
+     * @param transaction represents the transaction
+     * @param kit represents the kit to add
+     * @return whether the action was successful or not
+     */
+    public KitResult addKit(Transaction transaction, Kit kit) {
+        int res = transaction.getProperty(DSLContext.class)
+                .insertInto(KITPVP_KITS_OWNERSHIP,KITPVP_KITS_OWNERSHIP.USER_ID,KITPVP_KITS_OWNERSHIP.KIT_ID)
+                .values(userID,kit.getId())
+                .onDuplicateKeyUpdate()
+                .set(KITPVP_KITS_OWNERSHIP.KIT_ID, kit.getId())
+                .execute();
+        return new KitResult(res != 0);
+    }
+
+    //TODO verify if the record-counting part of this (res != 0) is accurate or needs to be flipped
+
+    /**
+     * Remove an existing kit from player's kit list
+     * @param transaction represents the transaction
+     * @param kit represents the kit to remove
+     * @return whether the action was successful or not
+     */
+    public KitResult removeKit(Transaction transaction, Kit kit) {
+        int res = transaction.getProperty(DSLContext.class)
+                .delete(KITPVP_KITS_OWNERSHIP)
+                .where(KITPVP_KITS_OWNERSHIP.USER_ID.eq(userID),KITPVP_KITS_OWNERSHIP.KIT_ID.eq(kit.getId()))
+                .execute();
+        return new KitResult(res != 0);
+    }
+
+    /**
+     * Gets all the kits the player owns
+     * @param transaction represents a...
+     * @return result containing all owned kits
+     */
+    public ListKitsResult getKits(Transaction transaction) {
```

Comment:
I do not think ListKitsResult is really necessary. Set<Kit> should do fine

Reply:
Doesn't that go against what you told me about the usage of Results?

Reply:
The reason I preferred result objects before is that a `boolean` is an ambiguous return type.

In this case Set<Kit> should be plenty clear as a return type.